### PR TITLE
docs(idempotency): split snippets, improve wording and lint examples

### LINKED ISSUE #1372: Refactor Idempotency code snippets

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #1372 

### Changes

Changes snippet fictitious names to real code snippet filenames for easier discovery and collaboration to improve/correct document snippets.

Some examples have also been refactored so that the codes are complete and there are no lint/pre-commit hooks issues.

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by heitorlessa** [comment]
WOOOT WOOOT!!! can't wait to review this. So exciting.

File: examples/idempotency/src/app_disabling_idempotency_utility.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from aws_lambda_powertools.utilities.idempotency import (
```
Comment (heitorlessa) [inline examples/idempotency/src/app_disabling_idempotency_utility.py]: is that for the testing section? we should have the name closely aligned with the section so it's easier to find.

Comment (leandrodamascena) [reply examples/idempotency/src/app_disabling_idempotency_utility.py]: @heitorlessa Make sense to move this to a specific folder and make changes in the name. Please, check it now.

**Comment by rubenfonseca** [comment]
Looking at this now

**Review by rubenfonseca — APPROVED** [review]
AWSome work!

**Pull request merged**